packages/db/src/models/base.ts

```diff
@@ -0,0 +1,42 @@
+import Knex from 'knex';
+
+import { tableNames } from '../tables';
+
+import { isRecordDefined } from '../database';
+
+class BasePgModel<ItemType, RecordType extends { cumulus_id: number }> {
+  readonly tableName: tableNames;
+
+  constructor({
+    tableName,
+  }: {
+    tableName: tableNames,
+  }) {
+    this.tableName = tableName;
+  }
+
+  get(
+    knexOrTrx: Knex | Knex.Transaction,
+    params: Partial<RecordType>
+  ) {
+    return knexOrTrx<RecordType>(this.tableName).where(params).first();
+  }
+
+  async exists(
+    knexOrTrx: Knex | Knex.Transaction,
+    params: Partial<RecordType>
+  ): Promise<boolean> {
+    return isRecordDefined(await this.get(knexOrTrx, params));
+  }
+
+  create(
+    knexOrTrx: Knex | Knex.Transaction,
+    item: ItemType
+  ) {
+    return knexOrTrx(this.tableName)
+      .insert(item)
+      .returning('cumulus_id');
```

Comment:
I went back and forth about whether we want to include this by default - but we almost always want to know the cumulus_id of the just created record

Reply:
I guess just because it feels YAGNI so I'm inclined to leave it until we decide that it needs to change. I looked into changing it, but using an array as the argument to `returning` changes the return structure so that you have to destructure the inserted row to get the cumulus_id instead of getting it directly from the query result. I can make the change if you think it's worthwhile, but this is the difference:

```
const [cumulusId] = await basePgModel.create(knex, { info });
```

vs using an array for `returning`:

```
const [insertedRow] = await basePgModel.create(knex, { info });
const { cumulus_id: cumulusId } = insertedRow;
```